src/Pipeline.h

```diff
@@ -498,9 +498,22 @@ class Pipeline {
      * of the appropriate size and binding them to the unbound
      * ImageParams. */
     // @{
+    void infer_input_bounds(const std::vector<int32_t> &sizes,
+                            const Target &target = get_jit_target_from_environment(),
+                            const ParamMap &param_map = ParamMap::empty_map());
+    HALIDE_ATTRIBUTE_DEPRECATED("Call infer_input_bounds() with an explicit vector<int> instead")
     void infer_input_bounds(int x_size = 0, int y_size = 0, int z_size = 0, int w_size = 0,
                             const Target &target = get_jit_target_from_environment(),
                             const ParamMap &param_map = ParamMap::empty_map());
+    // TODO: this is a temporary wrapper used to disambiguate the cases where
```

Comment:
Is this the full story? Or does this serve the single-int-arg case too and so we'll have to keep it?

Reply:
No: I have verified that code in the new form works correctly both in this PR form, and also if we delete both the deprecated method and this helper.

Reply:
Wait, what? I thought the {single_int} case was calling the wrong vector constructor?

Reply:
Yeah, that's what I thought too, but I apparently was wrong. 

The issue apparently is that `{1}` is a perfectly valid way to initialize an int (eg, `int foo{1};`); as a result, calling `infer_input_bounds({1});` bound to the deprecated method: with default arguments, there is effectively an overload of the form `infer_input_bounds(int x_size);`, and that's preferred since it doesn't have to construct anything. Adding the initializer_list version provides another layer of preference.

(I hope I have got all of this right... this part of C++ is just a nightmare of twisty little passages)